### PR TITLE
JAMES-4077 Back memory-app with Lucene

### PR DESCRIPTION
**Pull request by vttranlina**
jira https://issues.apache.org/jira/browse/JAMES-4077
resolve https://github.com/linagora/james-project/issues/5305

### CODE REVIEW

File: server/container/guice/memory/src/main/java/org/apache/james/modules/mailbox/MemoryMailboxModule.java [hunk 1]
```diff
@@ -99,6 +97,7 @@ protected void configure() {
         install(new MemoryDeadLetterModule());
         install(new MemoryQuotaModule());
         install(new MemoryQuotaSearchModule());
+        install(new LuceneSearchMailboxModule());
```
Comment (quantranhong1999) [inline server/container/guice/memory/src/main/java/org/apache/james/modules/mailbox/MemoryMailboxModule.java:100]: Not sure we want to use the File Directory (in `LuceneSearchMailboxModule`) for the memory app.

Memory app leaving some files on the host file system after testing sounds weird, no?

Likely we should use the Ram directory instead cf `LuceneMemorySearchMailboxModule`?

Comment (quantranhong1999) [reply server/container/guice/memory/src/main/java/org/apache/james/modules/mailbox/MemoryMailboxModule.java:100]: another reason: RAM is way faster than disk for testing. 

AFAIK we do use memory directory for Lucene is tests too.

**Review by quantranhong1999 — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Comment by quantranhong1999** [comment]
```java
Expecting actual throwable to be an instance of:
  java.lang.RuntimeException
but was:
  org.apache.james.StartUpChecksPerformer$StartUpChecksException: StartUpChecks got bad results: CheckResult{name=MailboxCapabilitiesForJMAP, resultType=BAD, description=Optional[Attachment Search support in MailboxManager is required by JMAP Module,Attachment file name Search support in MailboxManager is required by JMAP Module]}
```
seems related

**Comment by vttranlina** [comment]
a bit long explain here
JMAP StartUpChecksPerformer require MessageSearchIndex support capability:
```
SearchCapabilities.AttachmentFileName,
SearchCapabilities.Attachment
```

But the reality `LuceneMessageSearchIndex` does not support it now, (we already have another ticket for it)
If we still try to add new capabilies for `LuceneMessageSearchIndex` (for pass StartUpChecksPerformer), 
then another test will fail
eg:
```
ERROR] Errors: 
[ERROR]   LuceneMessageSearchIndexTest>AbstractMessageSearchIndexTest.searchShouldRetrieveMailByAttachmentFileName:1633 » UnsupportedSearch
[ERROR]   LuceneMessageSearchIndexTest>AbstractMessageSearchIndexTest.searchWithPDFAttachmentShouldReturnMailsWhenAttachmentContentMatches:1482 » UnsupportedSearch
[ERROR]   LuceneMessageSearchIndexTest>AbstractMessageSearchIndexTest.searchWithTextAttachmentShouldNotMatchMessageBody:1458 » UnsupportedSearch
[ERROR]   LuceneMessageSearchIndexTest>AbstractMessageSearchIndexTest.searchWithTextAttachmentShouldReturnMailsWhenAttachmentContentMatches:1443 » UnsupportedSearch
```

**Review by chibenwa — APPROVED** [review]

**Comment by vttranlina** [comment]
Not simple as my thought (just guice binding done)
a lot of test case for email query (memory test) failed, related to Lucense search 
`org.apache.james.jmap.rfc8621.memory.MemoryEmailQueryMethodNoViewTest`
```
listMailsBySystemKeywordShouldReturnOnlyMailsWithThisSystemKeyword(Flags, String, GuiceJamesServer)
[5] keywordFlag=$Forwarded, keywordName=$Forwarded
ccShouldFilterResultsWhenAddress(GuiceJamesServer)
toShouldFilterResultsWhenAddress(GuiceJamesServer)
listMailsNotBySystemKeywordShouldReturnOnlyMailsWithoutThisSystemKeyword(Flags, String, GuiceJamesServer)
bccShouldFilterResultsWhenAddress(GuiceJamesServer)
emailQueryShouldSupportTextFilterForTextBody(GuiceJamesServer)
emailQueryShouldSupportTextFilterForHeaders(GuiceJamesServer)
emailQueryShouldSupportTextFilterForHtmlBody(GuiceJamesServer)
emailQueryFilterByTextShouldIncludeAttachmentName(GuiceJamesServer)
emailQueryShouldSupportTextFilterForMultipartMessage(GuiceJamesServer)
fromShouldFilterResultsWhenAddress(GuiceJamesServer)
bodyFilterShouldMatchTextPlainOnAttachmentsFileName(GuiceJamesServer)
bodyFilterShouldMatchTextPlainOnAttachmentsContent(GuiceJamesServer)
```

**Review by Arsnael — APPROVED** [review]

**Comment by vttranlina** [comment]
squash

**Comment by vttranlina** [comment]
because the pr 
- https://github.com/apache/james-project/pull/2476

I will  rebase on that pr, and adapt it (no need disabled some test cases)

**Comment by Arsnael** [comment]
Can rebase

**Comment by vttranlina** [comment]
rebase

**Pull request merged**